odl/trafos/fourier.py

```diff
@@ -1365,13 +1366,25 @@ def _call_pyfftw(self, x, out, **kwargs):
         return out
 
     @property
+    def adjoint(self):
+        """The adjoint Fourier transform."""
+        if self.domain.exponent != 2.0 or self.range.exponent != 2.0:
+            raise NotImplementedError(
+                'Adjoint only implemented for exponent=2')
```

Comment:
My intuition (somewhat informed since I worked on that stuff in my PhD) is that you get the exact same expressions for the L^p case (1 <= p < infty) since the replacement of the inner product has the same form (integral), and the defining equation for the adjoint is the same.
The only notable difference is that the domain of the operator would be smaller, making it a closed operator only, no longer bounded (for p > 2), and the equality `<F f, g>_(L^q x L^p) = <f, F^* g>_(L^p x L^q)` would only hold for `f, g` such that `f in domain(F)` and `g in domain(F^*)`.

Anyway, regarding warnings I don't know. They are extremely annoying and interfere with screen output a lot. If at all, we would need a method to switch them off.